src/scripting/StelMainScriptAPI.cpp

```diff
@@ -342,6 +342,11 @@ QStringList StelMainScriptAPI::getAllTimezoneNames()
 
 void StelMainScriptAPI::screenshot(const QString& prefix, bool invert, const QString& dir, const bool overwrite, const QString &format)
 {
+	if (!qApp->property("allow_unsafe").toBool()) {
```

Comment:
Do I understand that this inhibits any screenshots requested by scripts unless the program is started with --unsafe? This means no screenshots (and complaints to us) for the average user, and permanently including --unsafe in the STEL_OPTS envvar (or Windows start link) for advanced users.

Reply:
> This means no screenshots (and complaints to us) for the average user

This does not block screenshot using for example Control+S. This part still works.

This only blocks screenshots through the scripting API. I guess the average user does not use this (?).

> Note that if dir is empty (as is common), storing goes to the preconfigured screenshot directory. IMO when dir=="" there should not be any danger.

This this should be improved by allowing safe calls to screenshot().

Reply:
> I guess the average user does not use this (?).

Periodic screenshots is a way to make a time lapse, and scripting seems like an ideal means to make such screenshots. Does "average user" not want time lapses?

Reply:
> IMO when dir=="" there should not be any danger.

Event in this case, it would be possible to writ to other directories through path traversal, for example using prefix=".././../foo".

Reply:
OK. This means we must add a test for "simple filename without devious traversal" for prefix. 
... done in 9ecd967f0a103b34f217c97c095ff1c6dfae8fb8

Other than that, running scripts which just store into the default screenshot directory should not pose a problem.

I can consider this as start of the method:

```
	if (dir.length()>0 && !qApp->property("allow_unsafe").toBool()) {
		qWarning() << "SCRIPTING CONFIGURATION ISSUE: You are trying to save a screenshot to a separate directory.";
				qWarning() << "  To enable this, pass the --unsafe flag";
				return;
	}
```

I still don't know what a QImage stored to some arbitrary path can break. Yes, you can overwrite an existing other file named like a supported image type, if you are entitled to such writing operation in that directory. Scenario example?

Reply:
> fill the disk (which is absolutely not mitigated).

For reference, I mentionned this for exhaustivity, not because I felt it was necessary to have some mitigation for this.